### PR TITLE
Feat (brevitas_examples/llm): Use dataloader in AWQ

### PR DESCRIPTION
**Pull request by pablomlago**
## Reason for this PR
AWQ was broken due to API mismatches with recent equalization wrapper changes and data loading patterns. Additionally, rotation matrix generation lacked a `generator` argument, forcing tests to monkey-patch for determinism. This PR fixes AWQ, adds CPU support, and improves reproducibility.

## Changes Made in this PR

**AWQ fixes** (`src/brevitas_examples/llm/llm_quant/awq/`):
- Updated `region.py` to use `EqualizationSourceWrapper`/`EqualizationSinkWrapper` instead of raw `EqualizationIndexes`.
- Switched `auto_scale.py` from `getfullargspec` to `inspect.signature`, skipping the first positional arg when forwarding kwargs.
- Changed `pre_quant.py` to accept a `DataLoader` instead of `DatasetToDevice`; made block output handling robust to non-tuple returns.
- Fixed `auto_clip.py` to clamp `group_size` and guard against zero-division when subsampling.
- Guarded all `.cuda()` calls with `torch.cuda.is_available()` for CPU compatibility.
**Rotation reproducibility** (`equalize.py`, `hadamard.py`):
- Added optional `generator: torch.Generator` to `random_orthogonal_matrix`, `random_hadamard_matrix`, and `_compute_rotations`.
**Other fixes**:
- Broadened RMSNorm detection in `_get_input_axis`/`_get_output_axis` to match any class name containing `"RMSNorm"`.
- Replaced `.view()` with `.reshape()` in `compute_perplexity` for non-contiguous tensor safety.
- Updated `main.py` to pass a `DataLoader` to both `compute_perplexity` and `apply_awq`.

**Tests**:
- Removed duplicated `_random_orthogonal_matrix` helper; tests now pass `generator` directly.
- Added `mock_compute_rotations` wrapper for deterministic PPLs in integration tests.
- Added new `awq_clip_scale` test case; updated expected PPL values.

## Testing Summary

- New `awq_clip_scale` integration test covering AWQ scale + clip.
- Rotation and perplexity tests updated with new expected values matching the generator-based approach.

## Risk Highlight

- [ ] This PR includes code from another work (please detail).
- [x] This PR contains API-breaking changes.
  - `apply_awq` now expects `calibration_loader: DataLoader` instead of `calibration_dataset: DatasetToDevice`.
- [ ] This PR depends on work in another PR (please provide links/details).
- [ ] This PR introduces new dependencies (please detail).
- [ ] There are coverage gaps not covered by tests.
- [ ] Documentation updates required in subsequent PR.

# Checklis
- [ ] Code comments added to any hard-to-understand areas, if applicable.
- [ ] Changes generate no new warnings.
- [x] Updated any relevant tests, if applicable.
- [ ] No conflicts with destination `dev` branch.
- [ ] I reviewed my own code changes.
- [ ] Initial CI/CD passing.
- [ ] 1+ reviews given, and any review issues addressed and approved.
- [ ] Post-review full CI/CD passing.

### CODE REVIEW

File: src/brevitas/graph/equalize.py [hunk 1]
```diff
@@ -481,7 +481,7 @@ def _get_input_axis(module: nn.Module) -> Optional[int]:
             return 0
         elif module.groups == module.out_channels:
             return 1
-    elif isinstance(module, (nn.LayerNorm, RMSNorm)):
+    elif isinstance(module, (nn.LayerNorm, RMSNorm)) or 'RMSNorm' in type(module).__name__:
```
Comment (Giuseppe5) [inline src/brevitas/graph/equalize.py:484]: Why do we need the extra check?

File: tests/brevitas_examples/test_llm_cases.py [hunk 2]
```diff
@@ -638,210 +650,210 @@ class LLMRotationOptimizationCases:
...
+                        "--use-distillation-loss",
+                        "True",
+                        "--max_steps",
+                        "2",
+                        "--per_device_train_batch_size",
+                        "1",
+                        "--gradient_accumulation_steps",
+                        "1"],
+                    "float_ppl": 32428.475,
+                    "quant_ppl": 32289.268,
```
Comment (Giuseppe5) [inline tests/brevitas_examples/test_llm_cases.py:850]: What's changed here and in the other new ppl values?

File: src/brevitas_examples/llm/llm_quant/awq/auto_clip.py [hunk 3]
```diff
@@ -161,7 +163,8 @@ def apply_clip(block_regions: List[RegionAWQ], clip_dict: Dict[int, torch.Tensor
         for name in region.sinks_names:
             if name in clip_dict:
                 sink = region.name_to_module[name]
-                sink.cuda()
+                if torch.cuda.is_available():
```
Comment (Giuseppe5) [inline src/brevitas_examples/llm/llm_quant/awq/auto_clip.py]: This could be a one liner

File: src/brevitas/graph/equalize.py [hunk 4]
```diff
@@ -1542,7 +1542,7 @@ def _apply_ort_device(tensor, ort, *args):
 
 
 # Adapted from https://github.com/facebookresearch/SpinQuant/blob/main/eval_utils/rotation_utils.py#L26
-def random_orthogonal_matrix(size):
+def random_orthogonal_matrix(size, generator: Optional[torch.Generator] = None):
```
Comment (Giuseppe5) [inline src/brevitas/graph/equalize.py:1545]: Let's discuss if we want to do this or just use the deterministic version `get_hadK`

**Review by Giuseppe5 — COMMENTED** [review]
Small updates and some questions.

Update the PR description.

File: src/brevitas/graph/equalize.py (hunk 4 above)
Comment (pablomlago) [reply src/brevitas/graph/equalize.py:1545]: I think it is orthogonal to that discussion.

File: src/brevitas/graph/equalize.py (hunk 1 above)
Comment (pablomlago) [reply src/brevitas/graph/equalize.py:484]: To detect the axis for `LlamaRMSNorm`, as these functions are called inside `AWQ`, and it seemed that doing the replacement by `RMSNorm` to make this function work was an overkill.
